### PR TITLE
APP-1433: Impersonation-support

### PR DESCRIPTION
**Pull request by hedvigoscar**

### CODE REVIEW

**Review by StylianosGakis — APPROVED** [review]

File: app/src/engineering/java/com/hedvig/app/feature/impersonation/ImpersonationReceiverActivity.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    object Event
+
+    private val _events = Channel<Event>(Channel.UNLIMITED)
+    val events = _events.receiveAsFlow()
+
+    init {
+        viewModelScope.launch {
+            if (authenticationTokenService.authenticationToken == null) {
+                authenticationTokenService.authenticationToken = "123"
```
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply app/src/engineering/java/com/hedvig/app/feature/impersonation/ImpersonationReceiverActivity.kt:119]: it's merely there to stuff something into the `Authorization`-header, as the underlying mutation won't work without a token (even though it never actually makes use of it 🤷 )

**Review by hugokallstrom — APPROVED** [review]
Looks good! 

Not sure I understand how I access this feature from the app though? 😅

**Comment by hedvigoscar** [comment]
Good question! To access it, you scan the QR-code in HOPE, which opens `hedvigengineering://impersonate?token={TOKEN}` - that URI launches this activity.

**Comment by hugokallstrom** [comment]
> Good question! To access it, you scan the QR-code in HOPE, which opens `hedvigengineering://impersonate?token={TOKEN}` - that URI launches this activity.

Awesome. Thanks!

**Pull request merged**